borg/archive.py

```diff
@@ -574,7 +574,7 @@ def process_file(self, path, st, cache, ignore_inode=False):
             status = 'A'  # regular file, added
         item = {
             b'path': safe_path,
-            b'link': st.st_nlink > 1,
+            b'link': st.st_nlink > 1,  # item is a hard link and has the chunks
```

Comment:
hardlink_master plus a note that we consider the first hardlink we encounter as master and save it with the data.